### PR TITLE
Use directory.build.props to centralize configuration

### PR DESCRIPTION
**Pull request by mfkl**
No functional change, except for adding `TreatWarningsAsErrors` to local dev and not only CI.

Existing work from https://github.com/G-Research/consuldotnet/pull/80

### CODE REVIEW

**Comment by mfkl** [comment]
Well, I guess this one is _still_ intermittently failing...

```
[xUnit.net 00:00:12.62]     Consul.Test.KVTest.KV_WatchList_Cancel [FAIL]
[xUnit.net 00:00:12.62]       A cancellation exception was not thrown when one was expected.
[xUnit.net 00:00:12.63]       Expected: True
[xUnit.net 00:00:12.63]       Actual:   False
[xUnit.net 00:00:12.63]       Stack Trace:
[xUnit.net 00:00:12.63]         /home/runner/work/consuldotnet/consuldotnet/Consul.Test/KVTest.cs(326,0): at Consul.Test.KVTest.KV_WatchList_Cancel()
[xUnit.net 00:00:12.63]         --- End of stack trace from previous location where exception was thrown ---
```

**Comment by mfkl** [comment]
Ah no, that's a new one. The one I fixed was `KV_WatchGet_Cancel`.

File: Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <PackageProjectUrl>https://github.com/G-Research/consuldotnet</PackageProjectUrl>
+    <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
+    <RepositoryType>git</RepositoryType>
+    <RepositoryUrl>https://github.com/G-Research/consuldotnet</RepositoryUrl>
+    <Authors>G-Research</Authors>
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+    <PublicSign>true</PublicSign>
+    <AssemblyOriginatorKeyFile>../assets/consuldotnet.snk</AssemblyOriginatorKeyFile>
+    <SignAssembly>true</SignAssembly>
+    <GeneratePackageOnBuild>false</GeneratePackageOnBuild>
```
Comment (marcin-krystianc) [inline Directory.Build.props:16]: I think that this is not necessary - it is false by default anyway.

**Review by marcin-krystianc — DISMISSED** [review]

File: Directory.Build.props (hunk 1 above)
Comment (mfkl) [reply Directory.Build.props]: OK

**Review by marcin-krystianc — APPROVED** [review]

**Review by jgiannuzzi — APPROVED** [review]
Great stuff, thanks @mfkl!

**Pull request merged**